src/validate.js

```diff
@@ -30,24 +29,45 @@ var resolutionSchema = exports.resolutionSchema = Joi.alternatives().try(
 
 var browsersSchema = exports.browsersSchema = Joi.array().min(1).unique().items(
   Joi.object().keys({
-    browserName: Joi.string().valid(['chrome', 'firefox']).required(),
+    browserName: Joi.string().valid('chrome', 'firefox').required(),
     includeRules: includeRulesSchema,
     excludeRules: excludeRulesSchema
   }),
   Joi.object().keys({
-    browserName: Joi.string().valid(['chrome', 'firefox', 'safari', 'microsoftedge', 'internet explorer']).required(),
-    version: Joi.string().required(),
+    browserName: Joi.string().valid('internet explorer').required(),
+    version: Joi.string().regex(/^[1-9]\d+$/).required(),
+    includeRules: includeRulesSchema,
+    excludeRules: excludeRulesSchema
+  }),
+  Joi.object().keys({
+    browserName: Joi.string().valid('chrome', 'firefox', 'safari', 'microsoftedge', 'internet explorer').required(),
+    version: Joi.string().regex(/^\d+\.\d+$/).required(),
     includeRules: includeRulesSchema,
     excludeRules: excludeRulesSchema
   })
-);
+).when('sauce.launchSauceConnect', { is: true, then: Joi.array().min(1).unique().items(
+  Joi.object().keys({
+    browserName: Joi.string().valid('chrome', 'firefox').required(),
+  }).forbidden(),
+  Joi.object().keys({
+    browserName: Joi.string().valid('internet explorer').required(),
+    version: Joi.string().regex(/^[1-9]\d+$/).required(),
+  }).forbidden(),
+  Joi.object().keys({
+    browserName: Joi.string().valid('chrome', 'firefox', 'safari', 'microsoftedge', 'internet explorer').required(),
+    version: Joi.string().regex(/^\d+\.\d+$/).required(),
+    includeRules: includeRulesSchema,
+    excludeRules: excludeRulesSchema
+  }).required()
+).error(new Error('Please make sure screener browsers are not set when launchSauceConnect flag is true and specify the version of the browser up to one digit after decimal point')) });
 
 var sauceSchema = exports.sauceSchema = Joi.object().keys({
   username: Joi.string().required(),
   accessKey: Joi.string().required(),
   maxConcurrent: Joi.number(),
+  launchSauceConnect: Joi.boolean(),
   extendedDebugging: Joi.boolean(),
-  tunnelIdentifier: Joi.string(),
+  tunnelIdentifier: Joi.string().when('launchSauceConnect', { is: true, then: Joi.forbidden().error(new Error('Can not specify tunnelIdentifier when sauce connect tunnel is being used')) }),
```

Comment:
Update error message to: `tunnelIdentifier cannot be set when launchSauceConnect flag is enabled`

Reply:
addressed